session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSessionRegistry.scala

```diff
@@ -81,7 +82,20 @@ final class CassandraSessionRegistry(system: ExtendedActorSystem) extends Extens
       executionContext: ExecutionContext): CassandraSession = {
     val sessionProvider = CqlSessionProvider(system, system.settings.config.getConfig(key.configPath))
     val log = Logging(system, classOf[CassandraSession])
-    new CassandraSession(system, sessionProvider, executionContext, log, metricsCategory = key.configPath, init)
+    new CassandraSession(
+      system,
+      sessionProvider,
+      executionContext,
+      log,
+      metricsCategory = key.configPath,
+      init,
+      onClose = () => sessions.remove(key))
+  }
+
+  def close(): Future[Done] = {
+    import system.dispatcher
+    val closing = sessions.values().asScala.map(_.close())
+    Future.sequence(closing).map(_ => Done)
```

Comment:
I don't think this should be here. If a user `close`s the `CassandraSessionRegistry` there is no way to start it again without starting a new actor system so you will leave the system in a weird state. Also logic that already have a handle to the session will get it closed from under their hands.

On the other hand, shouldn't the registry register something on system termination and then close all open sessions?

Reply:
My idea was to use this eg. from coordinated shutdown or actor system `whenTerminated`.
But to have it internal would be enough.